### DISCUSSION TITLE
Less FOIC

### DISCUSSION THREAD

**Discussion by wildfiremedia** [alpinejs/alpine#2898] [3. Feature ideas]
One interesting effect I have noticed that Twind use

```
<html hidden>
...
<script src="//unpkg.com/alpinejs" defer></script>
<script type="module" src="https://cdn.skypack.dev/twind/shim"></script>
</html>
```
Which makes FOIC Flash of Invisible Content - less likely noticeable when doing Javascript fetch() with JSON.

Compare to Alpine.js using, I still get a glaring delay FOIC with Javascript fetch.
```
[x-cloak] {display: none}
```

Could Twind's resolution removed `hidden` be implement in Alpine.js?

Doc reference:
https://twind.dev/handbook/the-shim.html#basic-usage

**Comment by SimoTod** [comment]
You can implement it by yourself with x-init, but... Basically you are hiding the whole page until your javascript executes (so even if the script is deferred, it's actually blocking the page rendering).
It might please the eye of developers/designers but, in the real world, it makes your site slower and it's likely not good SEO-wise.

**Reply by wildfiremedia** [reply]
Will find out on x-init.

To give more context, our site is primarily focused in Singapore for local users, so it's not a problem since it's at least sub-millisecond network latencies for VPS and CDN within Singapore and run on custom backend.
Before: 0.0057 (CLS for mobile and desktop)
After: 0 (for mobile and desktop)

But we could have an option to hide the whole page only for certain situations where we don't need SEO but prefer the experience of SPA like.

**Reply by SimoTod** [reply]
No worries, be aware that mobile is usually more complicate (slow/old devices), poor signal when not using a wifi connection, etc.
Technically, you can add a hidden attribute to your component (or the full page) and remove it when the fetch promise resolves. You don't even need x-init as I said before (it probably won't work well for your use case anyway since fetch is asynchronous so you won't know how long it takes).
